### PR TITLE
Document exported functions in agent.go

### PR DESCRIPTION
**Pull request by shaqq**
I wasn't sure how deep I should document `ReloadConfig`, since that could change over time and be quite complicated. Let me know what I should add/remove!

### CODE REVIEW

File: agent/agent.go [hunk 1]
```diff
@@ -3992,6 +3996,9 @@ func (a *Agent) loadLimits(conf *config.RuntimeConfig) {
 	a.config.RPCMaxBurst = conf.RPCMaxBurst
 }
 
+// ReloadConfig will atomically reload all configs from the given newCfg,
+// including all services, checks, tokens, metadata, dnsServer configs, etc.
+// It will also reload all ongoing watches.
 func (a *Agent) ReloadConfig(newCfg *config.RuntimeConfig) error {
```
Comment (shaqq) [inline agent/agent.go]: This method seemed like a doozy. I'm sure there's some context that should be provided here (perhaps historical? or design related?). Let me know what it is and I'm happy to update the PR!

File: agent/agent.go [hunk 2]
```diff
@@ -3644,7 +3648,7 @@ func (a *Agent) loadServices(conf *config.RuntimeConfig) error {
 		}
 	}
 
-	for serviceID, _ := range persistedServiceConfigs {
+	for serviceID := range persistedServiceConfigs {
```
Comment (shaqq) [inline agent/agent.go]: `golint` told me to do this, but let me know if this should be put back. i understand wanting to hint to the reader that `_` is a possible value.

**Comment by i0rek** [comment]
I missed the conflict, could you take care of that one?

**Comment by shaqq** [comment]
@i0rek Done! Do you need me to squash my commits as well, or can you do that with Github's squash & merge?

**Comment by i0rek** [comment]
Thanks again! And yes, we always squash and merge which takes care of all the squashing and merging ;).

**Pull request merged**